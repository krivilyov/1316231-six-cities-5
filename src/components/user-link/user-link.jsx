import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {getIsAuthorizedStatus} from "../../store/selectors";
import {AppRoute} from "../../const";
import {Link} from "react-router-dom";
import UserLinkContent from "../user-link-content/user-link-content";

class UserLink extends PureComponent {
  constructor(props) {
    super(props);
  }

  _getLink(currentRoute, isAuthorizedStatus) {
    switch (true) {
      case currentRoute !== AppRoute.FAVORITES && isAuthorizedStatus:
        return AppRoute.FAVORITES;
      case currentRoute !== AppRoute.LOGIN && !isAuthorizedStatus:
        return AppRoute.LOGIN;
      default:
        return false;
    }
  }

  render() {
    const {appRoute, isAuthorizedStatus} = this.props;
    const link = this._getLink(appRoute, isAuthorizedStatus);

    return appRoute !== AppRoute.FAVORITES && isAuthorizedStatus || appRoute !== AppRoute.LOGIN && !isAuthorizedStatus ? (
      <Link className="header__nav-link header__nav-link--profile"
        to={link}
      >
        <UserLinkContent />
      </Link>
    ) : (
      <a className="header__nav-link header__nav-link--profile header__nav-link--inactive">
        <UserLinkContent />
      </a>
    );
  }
}

UserLink.propTypes = {
  appRoute: PropTypes.string.isRequired,
  isAuthorizedStatus: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  isAuthorizedStatus: getIsAuthorizedStatus(state),
});

export {UserLink};
export default connect(mapStateToProps)(UserLink);
