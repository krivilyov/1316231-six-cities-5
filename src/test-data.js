export const testInitialState = {
  COMMON: {
    activeCity: `Amsterdam`,
    offers: [],
    reviews: [],
    sortingType: `popular`,
    mouseOverOfferId: null,
    changedBookmarkOffer: {},
    relatedOffers: [],
    bookmarkOffers: [],
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    authInfo: {},
  },
};
