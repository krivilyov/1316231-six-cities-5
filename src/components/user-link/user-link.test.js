import React from "react";
import renderer from "react-test-renderer";
import {UserLink} from "./user-link";
import configureMockStore from "redux-mock-store";
import {testInitialState} from "../../test-data";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

const mockStore = configureMockStore();
const store = mockStore(testInitialState);

describe(`UserLink should render correctly`, () => {
  it(`UserLink should render correctly when route not /login and isAuthorizedStatus={false}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <UserLink
                appRoute={`/`}
                isAuthorizedStatus={false}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`UserLink should render correctly when route not /favorites and isAuthorizedStatus={true}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <UserLink
                appRoute={`/`}
                isAuthorizedStatus={true}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
