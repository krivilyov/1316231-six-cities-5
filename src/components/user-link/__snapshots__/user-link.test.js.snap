// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLink should render correctly UserLink should render correctly when route not /favorites and isAuthorizedStatus={true} 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/favorites"
  onClick={[Function]}
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  />
  <span
    className="header__login"
  >
    Sign in
  </span>
</a>
`;

exports[`UserLink should render correctly UserLink should render correctly when route not /login and isAuthorizedStatus={false} 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/login"
  onClick={[Function]}
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  />
  <span
    className="header__login"
  >
    Sign in
  </span>
</a>
`;
